util/unittest/memblocktest.cpp

```diff
@@ -0,0 +1,58 @@
+/****************************************************************************
+ *
+ * Copyright (c) 2024 plgd.dev s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"),
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "util/oc_memb.h"
+
+#include <gtest/gtest.h>
+
+struct test_data_t
+{
+  int a;
+  int b;
+};
+
+class TestMemoryBlock : public testing::Test {
+public:
+};
+
+TEST_F(TestMemoryBlock, Init)
+{
+  OC_MEMB_LOCAL(oc_test_data, test_data_t, 42);
+  EXPECT_EQ(oc_test_data.size, sizeof(test_data_t));
+  oc_memb_init(&oc_test_data);
+#ifdef OC_DYNAMIC_ALLOCATION
+  EXPECT_EQ(oc_test_data.num, 0);
+#else  /* !OC_DYNAMIC_ALLOCATION */
+  EXPECT_EQ(oc_test_data.num, 42);
+#endif /* OC_DYNAMIC_ALLOCATION */
+}
+
+TEST_F(TestMemoryBlock, Alloc)
+{
+  // TODO
```

Comment:
@coderabbitai add the unit tests here

Reply:
@coderabbitai provide code snippets or commitable suggestions for all the suggestions you made previously for this file